### DISCUSSION TITLE
Data-driven documentation-ing

### DISCUSSION THREAD

**Discussion by JJ** [perl6/doc#4141] [Ideas]
TL;DR: data-mine GitHub looking for usage of different Raku features.

There's moment now where there are thousands of Raku modules published, and lots of other repos that use Raku. Using GitHub search, we could look for them and then log their usage of specific syntactic features, classes, and class methods. With this information, there are several things that can be done
* Try and match usage with documentation. If a specific thing is used widely, it should be well documented; all issues related to that specific feature solved and, in general, prioritized over other features.
* Link from documentation to examples of using them "in the wild", or harvest it for examples.

I general, anyway, it's interesting to find out how many repositories are there, how they grow, and of course help them whenever possible.

**Comment by Altai-man** [comment]
This sounds like a topic for a research paper, but not an action-able ticket, no?

**Comment by JJ** [comment]
Well, the second item is quite actionable, and can be done by just using the GitHub search engine. Doing systematically is something else, but I have always thought that working in the documentation only reactively is the wrong approach.

**Comment by JJ** [comment]
Also, don't see anything wrong in research stuff in a language that started basically as a research project in virtual machines...

**Comment by MasterDuke17** [comment]
This sounds like it would make a nice GSoC proposal, especially once the RakuAST work is finished.

**Comment by JJ** [comment]
I hadn't thought that, but yes. Even Season of Docs, next year...
